java/src/main/c++/jni_spark_vw.cc

```diff
@@ -415,62 +498,70 @@ JNIEXPORT void JNICALL Java_org_vowpalwabbit_spark_VowpalWabbitExample_setLabel(
   }
 }
 
-// re-use prediction conversation methods
-jobject multilabel_predictor(example* vec, JNIEnv* env);
-jfloatArray scalars_predictor(example* vec, JNIEnv* env);
-jobject action_scores_prediction(example* vec, JNIEnv* env);
-jobject action_probs_prediction(example* vec, JNIEnv* env);
-
-JNIEXPORT jobject JNICALL Java_org_vowpalwabbit_spark_VowpalWabbitExample_getPrediction(JNIEnv* env, jobject exampleObj)
+JNIEXPORT void JNICALL Java_org_vowpalwabbit_spark_VowpalWabbitExample_setDefaultLabel(JNIEnv* env, jobject exampleObj)
 {
   INIT_VARS
 
-  jclass predClass;
-  jmethodID ctr;
-  switch (all->l->pred_type)
+  try
   {
-    case prediction_type_t::scalar:
-      predClass = env->FindClass("org/vowpalwabbit/spark/prediction/ScalarPrediction");
-      CHECK_JNI_EXCEPTION(nullptr);
-
-      ctr = env->GetMethodID(predClass, "<init>", "(FF)V");
-      CHECK_JNI_EXCEPTION(nullptr);
-
-      return env->NewObject(predClass, ctr, VW::get_prediction(ex), ex->confidence);
-
-    case prediction_type_t::prob:
-      predClass = env->FindClass("java/lang/Float");
-      CHECK_JNI_EXCEPTION(nullptr);
+    all->p->lp.default_label(&ex->l);
+  }
+  catch (...)
+  {
+    rethrow_cpp_exception_as_java_exception(env);
+  }
+}
 
-      ctr = env->GetMethodID(predClass, "<init>", "(F)V");
-      CHECK_JNI_EXCEPTION(nullptr);
+JNIEXPORT void JNICALL Java_org_vowpalwabbit_spark_VowpalWabbitExample_setContextualBanditLabel(
+    JNIEnv* env, jobject exampleObj, jint action, jdouble cost, jdouble probability)
+{
+  INIT_VARS
 
-      return env->NewObject(predClass, ctr, ex->pred.prob);
+  try
+  {
+    CB::label* ld = &ex->l.cb;
+    CB::cb_class f;
 
-    case prediction_type_t::multiclass:
-      predClass = env->FindClass("java/lang/Integer");
-      CHECK_JNI_EXCEPTION(nullptr);
+    f.action = (uint32_t)action;
+    f.cost = (float)cost;
+    f.probability = (float)probability;
 
-      ctr = env->GetMethodID(predClass, "<init>", "(I)V");
-      CHECK_JNI_EXCEPTION(nullptr);
+    ld->costs.push_back(f);
+  }
+  catch (...)
+  {
+    rethrow_cpp_exception_as_java_exception(env);
+  }
+}
 
-      return env->NewObject(predClass, ctr, ex->pred.multiclass);
+JNIEXPORT void JNICALL Java_org_vowpalwabbit_spark_VowpalWabbitExample_setSharedLabel(JNIEnv* env, jobject exampleObj)
+{
+  INIT_VARS
 
-    case prediction_type_t::scalars:
-      return scalars_predictor(ex, env);
+  try
+  {
+    // https://github.com/VowpalWabbit/vowpal_wabbit/blob/master/vowpalwabbit/parse_example_json.h#L437
+    CB::label* ld = &ex->l.cb;
+    CB::cb_class f;
 
-    case prediction_type_t::action_probs:
-      return action_probs_prediction(ex, env);
+    f.partial_prediction = 0.;
+    f.action = (uint32_t)uniform_hash("shared", 6, 0);

```

Comment:
maybe add comment referencing that 6 is length of shared and 0 is the seed. This is an ugly api but not part of this PR.

Reply:
totally optional though, not sure how many people run into this code or whats the probability of someone changing shared to another string

Reply:
Not sure why the action must be this value... But this was copied from elsewhere